### ISSUE TITLE
net-wireless/wifite2 : No WPS networks show up in list

### ISSUE THREAD

**Issue by yuri-sevatz** [pentoo/pentoo-overlay#139]
I think i unearthed a runtime conflict between the following packages:

-net-wireless/wifite2
-net-wireless/reaver-wps-fork-t6x

In net-wireless/reaver-wps-fork-t6x (Commit 63c5c02bfc44e732399118d844906b0708e8e28c in upstream's source code), it looks like they changed the meaning of the '-C' flag from net-wireless/reaver's old "**Don't validate FCS**" to the new "**Do validate FCS**".

```
-    set_validate_fcs(1);
+    set_validate_fcs(0); //mod by flatr0ze
```

```
             case 'C':
-                set_validate_fcs(0);
+                set_validate_fcs(1); //mod by flatr0ze
```

Yet in the python version of net-wireless/wifite2, the code that still passes '-C' does not seem to follow suit, despite the version of wifilite2 shipped with pentoo insisting on using the forked version of reaver-wps-fork-t6x:

```
        command = [
            'wash',
            '-f', capfile, # Path to cap file
            '-C'            # Ignore Frame Check Sum errors
        ]
```

This looks like it's why nothing shows up as WPS-capable in wifite2's scan output, because wifite2's command ends up seeing nothing but wash output that looks like this:

> [!] Found packet with bad FCS, skipping...
> [!] Found packet with bad FCS, skipping...
> [!] Found packet with bad FCS, skipping...

So, either (A) the original commit changing the meaning of '-C' needs to be reverted in the forked version of reaver to maintain runtime compatibility with wifite2, or (B) pentoo's insistence on using the forked version of reaver needs to come with a corresponding patch to wash.py to remove the '-C' flag.

(I suspect the reason wifite2 does not want to do validation on FCS is because Linux driver code in doesn't pass the FCS information up to airodump.  Everyone says you can get it from Kismet, because Kismet rebuilds the FCS, but not from airodump-ng!)

**Comment by blshkv** [comment]
We don't ship wifite2 by default. It is in the tree for people who know what they are doing.
I remember you mentioned not to use wifite2 and wait for it to merge back to wifite. So we are waiting..

**Comment by blshkv** [comment]
it sounds like net-wireless/reaver-wps-fork bug. Could you file an upstream bug report? I'll be happy to bump it after that.

**Comment by blshkv** [comment]
These 2 projects such a mess. There are no stable releases, no properly defined dependences. Just some random forks where authors free to change meanings of parameters to exact opposite.

**Comment by yuri-sevatz** [comment]
I've opened a question with wifite2 in the WPS wifite2 issue (above), asking specifically which version of reaver wifite2 is meant to use.

From what I understand, the pixie attack was added in one of the forks.  If we can prove the fork required has the broken '-C' flag, then we can pass the bug onto wifite2.

By the looks of it, wifite v1 used the original version of reaver, but that version didn't support the WPS pixie attack (added in one of the reaver forks).  If that's the version they require, then we can blame wifite2 and expect them to fix upstream.
